docs/notes/shutdown.txt

```diff
@@ -0,0 +1,68 @@
+#
+# Licensed to the Apache Software Foundation (ASF) under one
+# or more contributor license agreements.  See the NOTICE file
+# distributed with this work for additional information
+# regarding copyright ownership.  The ASF licenses this file
+# to you under the Apache License, Version 2.0 (the
+# "License"); you may not use this file except in compliance
+# with the License.  You may obtain a copy of the License at
+#
+#   http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing,
+# software distributed under the License is distributed on an
+# "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+# KIND, either express or implied.  See the License for the
+# specific language governing permissions and limitations
+# under the License.
+#
+
+
+
+qdrouterd shutdown procedure
+============================
+
+
+How does the router shut down when interrupted with ctrl-C ?
+
+
+  1. ctrl-c is caught by server.c::handle().
+     it comes packaged by proton as PN_PROACTOR_INTERRUPT.
```

Comment:
Ctrl-c, i.e. SIGINT, is caught in main.c::signal_handler(). The handler calls qd_server_stop(). That results in delivering an interrupt into the Proton proactor. When proactor gets around to noticing that happened, it will emit `PN_PROACTOR_INTERRUPT`, which gets handled in server.c::handle().

Reply:
excellent -- thanks!